### PR TITLE
fix: prevent file traversal using helm file values param and application details api

### PR DESCRIPTION
**Pull request by alexmt**
Signed-off-by: Alexander Matyushentsev [AMatyushentsev@gmail.com](mailto:AMatyushentsev@gmail.com)

PR introduces missing checks in repo server API that prevents file traversal attack:

adds missing checks of helm value files in repository.GetAppDetails API
adds missing checks of helm fileParameters in repository.GenerateManifest API

### CODE REVIEW

File: reposerver/repository/repository.go [hunk 1]
```diff
@@ -1511,7 +1361,16 @@ func populateHelmAppDetails(res *apiclient.RepoAppDetailsResponse, appPath strin
...
 		return err
 	}
-	params, err := h.GetParameters(selectedValueFiles)
+	var resolvedSelectedValueFiles []pathutil.ResolvedFilePath
+	// drop not allowed values files
+	for _, file := range selectedValueFiles {
+		if resolvedFile, _, err := pathutil.ResolveFilePath(appPath, repoRoot, file, q.GetValuesFileSchemes()); err == nil {
+			resolvedSelectedValueFiles = append(resolvedSelectedValueFiles, resolvedFile)
+		} else {
+			log.Debugf("Values file %s is not allowed: %v", file, err)
```
Comment (jessesuen) [inline reposerver/repository/repository.go:1370]: Is there a reason this is a log message but not an error?

**Review by jessesuen — APPROVED** [review]
One question, but otherwise LGTM

File: reposerver/repository/repository.go [hunk 2]
```diff
@@ -807,18 +658,17 @@ func helmTemplate(appPath string, repoRoot string, env *v1alpha1.Env, q *apiclie
 		}
 
 		if appHelm.Values != "" {
-			file, err := ioutil.TempFile("", "values-*.yaml")
+			rand, err := uuid.NewRandom()
 			if err != nil {
 				return nil, err
 			}
-			p := file.Name()
+			p := path.Join(os.TempDir(), rand.String())
```
Comment (crenshaw-dev) [inline reposerver/repository/repository.go:665]: Could we move `createTempDir` from util/helm/client to util/io and use it here?

**Review by crenshaw-dev — APPROVED** [review]
LGTM! One small refactor opportunity if you care to take it.

File: reposerver/repository/repository.go (hunk 1 above)
Comment (alexmt) [reply reposerver/repository/repository.go:1423]: I did it to keep API backward compatible. App details API was not failing before if the user provide a file that does not exist, so I decided to keep this behavior for now.

**Comment by rajivml** [comment]
@alexmt does this PR has missing fixes which tackles this CVE ? https://github.com/argoproj/argo-cd/security/advisories/GHSA-63qx-x74g-jcr7

we are upgraded our argo to 2.2.5 recently to fix this CVE and was wondering if we have to upgrade to 2.2.6 instead to completely fix this ?

**Pull request merged**